python/tests/test_topology.py

```diff
@@ -7843,3 +7843,41 @@ def test_failure_with_migrations(self):
         self.assertRaises(ValueError, ts.ltrim)
         self.assertRaises(ValueError, ts.rtrim)
         self.assertRaises(ValueError, ts.trim)
+
+
+class TestMissingData(unittest.TestCase):
+    """
+    Test various aspects of missing data functionality
+    """
+
+    # TODO tests for missing data currently sparse: more tests should go here
+
+    def ts_missing_middle(self):
+        # Simple ts with sample 0 missing a middle section
+        ts = msprime.simulate(4, mutation_rate=1, recombination_rate=4, random_seed=2)
+        tables = ts.dump_tables()
+        tables.edges.clear()
+        # mark the middle as missing
+        for e in ts.tables.edges:
+            if e.child == 0:
+                if e.left == 0.0:
+                    missing_from = e.right
+                elif e.right == 1.0:
+                    missing_to = e.left
+                else:
+                    continue  # omit this edge => node is isolated
+            tables.edges.add_row(e.left, e.right, e.parent, e.child)
+        # Check we have non-missing to L & R
+        self.assertTrue(0.0 < missing_from < 1.0)
+        self.assertTrue(0.0 < missing_to < 1.0)
+        return tables.tree_sequence(), missing_from, missing_to
+
+    def test_is_isolated(self):
```

Comment:
I know this is a very simple method - but it's always good to test a couple of error cases, e.g. a negative index, a index bigger than the number of samples, an index of the wrong type just to check we get the right error messages.